src/aosm/azext_aosm/common/utils.py

```diff
@@ -125,6 +132,151 @@ def check_tool_installed(tool_name: str) -> None:
+        logger.debug(
+            "Output from %s: %s. Error: %s",
+            log_cmd,
+            called_process.stdout,
+            called_process.stderr,
+        )
+
+        return called_process.stdout
+    except subprocess.CalledProcessError as error:
+        all_output: str = (
+            f"Command: {' '.join(log_cmd)}\n"
+            f"stdout: {error.stdout}\n"
+            f"stderr: {error.stderr}\n"
+            f"Return code: {error.returncode}"
+        )
+        logger.debug("The following command failed to run:\n%s", all_output)
+        # Raise the error without the original exception, which may contain secrets.
+        raise CLIError(all_output) from None
+
+
+def clean_registry_name(registry_name: str) -> str:
+    """Remove https:// from the registry name."""
+    return registry_name.replace("https://", "")
+
+
+def push_image_from_local_registry_to_acr(
+    target_acr: str,
+    target_image: str,
+    target_username: str,
+    target_password: str,
+    local_docker_image: str,
+) -> None:
+    """
+    Push image to target registry using docker push. Requires docker.
+
+    :param target_acr: name of the target Azure Container registry
+        e.g. targetacr.azurecr.io
+    :type target_acr: str
+    :param target_image: name of the target image (namespace/repository:tag)
+        e.g. namespace/nginx:1.0.0
+    :type target_image: str
+    :param target_username: username for the target ACR
+    :type target_username: str
+    :param target_password: password for the target ACR
+    :type target_password: str
+    :param local_docker_image: name and tag of the source image on local registry
+        e.g. uploadacr.azurecr.io/samples/nginx:stable
+    :type local_docker_image: str
+    """
+
+    target = f"{target_acr}/{target_image}"
+    logger.debug("Target ACR: %s", target)
+
+    # To push the image to the target registry, we need to tag the source image
+    logger.info("Tagging source image %s as %s", local_docker_image, target)
+    tag_image_cmd = [
+        str(shutil.which("docker")),
+        "tag",
+        local_docker_image,
+        target,
+    ]
+    call_subprocess_raise_output(tag_image_cmd)
+
+    login_to_artifact_store_registry(target_acr, target_username, target_password)
+
+    try:
+        logger.info("Pushing target image %s using docker push", target)
+        push_target_image_cmd = [
+            str(shutil.which("docker")),
+            "push",
+            target,
+        ]
+        call_subprocess_raise_output(push_target_image_cmd)
+    except CLIError as error:
+        logger.debug(error, exc_info=True)
+        raise ClientRequestError(
+            f"Failed to push {local_docker_image} to {target_acr}."
+        ) from error
+    finally:
+        docker_logout_cmd = [
+            str(shutil.which("docker")),
+            "logout",
+            target_acr,
+        ]
+        call_subprocess_raise_output(docker_logout_cmd)
+
+
+def login_to_artifact_store_registry(
```

Comment:
Would presumably move with `push_image_from_local_registry_to_acr()`, if you took that suggestion.